test/integration/redis_test.rb

```diff
@@ -9,9 +9,38 @@ def setup
 
     def test_redis_runner
       output = `ruby -Itest/fixtures test/fixtures/redis-runner.rb`.lines.map(&:strip).last
-      assert_equal '4 runs, 0 assertions, 0 failures, 0 errors, 0 skips', output
+      assert_equal '4 tests, 2 assertions, 1 failures, 1 errors, 1 skips', output
       output = `ruby -Itest/fixtures test/fixtures/redis-runner.rb retry`.lines.map(&:strip).last
-      assert_equal '4 runs, 0 assertions, 0 failures, 0 errors, 0 skips', output
+      assert_equal '4 tests, 2 assertions, 1 failures, 1 errors, 1 skips', output
+    end
+
+    def test_redis_reporter
+      summary = Minitest::Reporters::RedisReporter::Summary.new(
+        redis: @redis,
+        build_id: 1,
+      )
+
+      output = `ruby -Itest/fixtures test/fixtures/redis-runner.rb`.lines.map(&:strip).last
+      assert_equal '4 tests, 2 assertions, 1 failures, 1 errors, 1 skips', output
+
+      io = StringIO.new
+      summary.report(io: io)
+      report = [
+        "Ran 4 tests, 2 assertions, \e[31m1 failures, 1 errors,\e[0m \e[33m1 skips\e[0m in 0.00s (aggregated).",
+        "FAIL ATest#test_bar",
+        "Expected false to be truthy.",
+        "    test/fixtures/dummy_test.rb:9:in `test_bar'",
+        "    lib/ci/queue/redis/worker.rb:32:in `poll'",
+        "",
+        "ERROR BTest#test_bar",
+        "TypeError: String can't be coerced into Fixnum",
+        "    test/fixtures/dummy_test.rb:19:in `+'",
+        "    test/fixtures/dummy_test.rb:19:in `test_bar'",
+        "",
+        "",
+      ].join("\n")
```

Comment:
I prefer `<<~HEREDOC` syntax for this, if you're using Ruby 2.2 or up. If not, https://github.com/Shopify/buildkite-runner/blob/master/test/lib/command_output_helpers.rb#L60-L63 will do the same.